sdk/tables/azure-tables/src/TableClient.ts

```diff
@@ -0,0 +1,155 @@
+  MergeEntityResponse,
+  GetAccessPolicyOptions,
+  GetAccessPolicyResponse,
+  SignedIdentifier,
+  SetAccessPolicyResponse
+} from "./generatedModels";
+
+/**
+ * A TableClient represents a Client to the Azure Tables service allowing you
+ * to perform operations on a single table.
+ */
+export class TableClient {
+  private client: TableServiceClient;
+  private tableName: string;
+
+  /**
+   * Initializes a new instance of the TableClient class.
+   * @param url The URL of the service account that is the target of the desired operation.
+   * @param tableName The name of the table
+   * @param options The parameter options.
+   */
+  // eslint-disable-next-line @azure/azure-sdk/ts-naming-options
+  constructor(url: string, tableName: string, options?: TableServiceClientOptions) {
+    this.client = new TableServiceClient(url, options);
+    this.tableName = tableName;
+  }
+
+  /**
+   * Returns a single entity in the table.
+   * @param partitionKey The partition key of the entity.
+   * @param rowKey The row key of the entity.
+   * @param options The options parameters.
+   */
+  getEntity(
+    partitionKey: string,
+    rowKey: string,
+    options?: GetEntityOptions
+  ): Promise<GetEntityResponse> {
+    return this.client.getEntity(this.tableName, partitionKey, rowKey, options);
+  }
+
+  /**
+   * Queries entities in the table.
+   * @param query The OData query parameters.
+   * @param options The options parameters.
+   */
+  listEntities(
+    // eslint-disable-next-line @azure/azure-sdk/ts-naming-options
+    query?: QueryOptions,
+    options?: ListEntitiesOptions
+  ): Promise<ListEntitiesResponse> {
+    return this.client.listEntities(this.tableName, query, options);
+  }
+
+  /**
+   * Insert entity in the table.
+   * @param entity The properties for the table entity.
+   * @param options The options parameters.
+   */
+  createEntity(entity?: Entity, options?: CreateEntityOptions): Promise<CreateEntityResponse> {
+    return this.client.createEntity(this.tableName, entity, options);
+  }
+
+  /**
+   * Deletes the specified entity in the table.
+   * @param partitionKey The partition key of the entity.
+   * @param rowKey The row key of the entity.
+   * @param ifMatch Match condition for an entity to be deleted. If specified and a matching entity is
+   *                not found, an error will be raised. To force an unconditional delete, set to the wildcard character
+   *                (*).
+   * @param options The options parameters.
+   */
+  deleteEntity(
+    partitionKey: string,
+    rowKey: string,
+    ifMatch: string,
+    options?: DeleteEntityOptions
+  ): Promise<DeleteEntityResponse> {
+    return this.client.deleteEntity(this.tableName, partitionKey, rowKey, ifMatch, options);
+  }
+
+  /**
+   * Update entity in the table.
+   * @param partitionKey The partition key of the entity.
+   * @param rowKey The row key of the entity.
+   * @param entity The properties for the table entity.
```

Comment:
In this operation, `entity` would be the "updated" entity that we'd like Table service to save right? Do you think we can be a bit more descriptive about this `entity` parameter?